core/domain/learner_progress_services.py

```diff
@@ -616,3 +648,118 @@ def get_collection_summary_dicts(collection_summaries):
         })
 
     return summary_dicts
+
+
+def get_activity_progress(user_id):
```

Comment:
Do you have a test (or more) for this? Try to set up a somewhat complicated case involving multiple entities with explorations being deleted etc.